### PR TITLE
fix: ensure absolute directories in glob patterns are leading prior to removal

### LINKED ISSUE #23361: 10.5.0 - "Your supportFile is missing or invalid" even when hardcoded in the config
### Current behavior

When we upgrade to Cypress 10.5.0 and run our E2E test suite through our pipelines, we get an error on each of docker containers:

> cypress_1  | Your project does not contain a default supportFile. We expect a file matching cypress/support/e2e.{js,jsx,ts,tsx} to exist.
> cypress_1  | 
> cypress_1  | If a support file is not necessary for your project, set supportFile to false.
> cypress_1  | 
> cypress_1  | https://on.cypress.io/support-file-missing-or-invalid

Even when we hardcode the location of the support file into the config, we get a similar error:

> cypress_2  | Your supportFile is missing or invalid: cypress/support/e2e.js
> cypress_2  | 
> cypress_2  | The supportFile must be a .js, .ts, .coffee file or be supported by your preprocessor plugin (if configured).
> cypress_2  | 
> cypress_2  | Fix your support file, or set supportFile to false if a support file is not necessary for your project.
> cypress_2  | 
> cypress_2  | If you have just renamed the extension of your supportFile, restart Cypress.
> cypress_2  | 
> cypress_2  | https://on.cypress.io/support-file-missing-or-invalid

When we downgrade to Cypress 10.4.0 - works again

### Desired behavior

Tests should just run as they did for 10.4.0

### Test code to reproduce

Not sure I can provide code - we didn't change anything. Initially, our configs did NOT have the supportFile location set

This is how our directory is structured - which is the default way, I believe:

Screenshot 2022-08-16 at 10 51 49

### Cypress Version

10.5.0

### Node version

16.16.0

### Operating System

Docker - running the Cypress included image for 10.5.0

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by tbiethman**
Closes #23361

### User facing changelog
 

* Fixes an issue where absolute path glob manipulation was incorrectly breaking supportFile/spec glob patterns that contained deep instances of the working directory.

### Additional details

The logic introduced with https://github.com/cypress-io/cypress/pull/23296 to broadly remove the project directory from our glob patterns was a little over-zealous. If the project directory was sufficiently simple and also found within the projects subdirectories (e.g. project located at `/e2e`, with a `--spec` value of `/e2e/cypress/e2e/*.cy.js`), the logic would behave incorrectly by replacing the nested instance of the directory (`cypress/e2e/*.cy.js` -> `cypress/.*.cy.js`). The same could be true for supportFile lookups, which go through the same logic.

This change ensures that we only remove the working directory from the glob pattern if it is in the leading position.

### PR Tasks

- [X] Have tests been added/updated?
- [X] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by emilyrohrbough — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

File: packages/data-context/src/sources/FileDataSource.ts [hunk 1]
```diff
@@ -36,7 +36,11 @@ export class FileDataSource {
       // The working directory path may include characters that conflict with glob
       // syntax (brackets, parentheses, etc.) and cause our searches to inadvertently fail.
       // We scope our search to the working directory using the `cwd` globby option.
-      return globPattern.replace(cwd, '.')
+      if (globPattern.startsWith(cwd)) {
+        return globPattern.replace(cwd, '.')
+      }
```
Comment (mjhenkes) [inline packages/data-context/src/sources/FileDataSource.ts:41]: A root directory of `/cypress/e2e/`, would still break right? `/cypress/e2e/cypress/e2e/*.cy.js` -> `*.cy.js`

Comment (mjhenkes) [reply packages/data-context/src/sources/FileDataSource.ts:41]: Is there any way you can just replace the first instance of the cwd?

Comment (tbiethman) [reply packages/data-context/src/sources/FileDataSource.ts:41]: Using `replace` (vs. `replaceAll`), only the first instance will be removed: `/cypress/e2e/cypress/e2e/*.cy.js` -> `./cypress/e2e/*.cy.js`

Comment (mjhenkes) [reply packages/data-context/src/sources/FileDataSource.ts:41]: So how did we see the issue initially? does this get called multiple times?

Comment (tbiethman) [reply packages/data-context/src/sources/FileDataSource.ts:41]: So in the issue, there's a working directory of `/e2e`, and a glob pattern of `cypress/support/e2e.{js,ts}`. Without the startsWith check, we get `cypress/support..{js.ts}`, which is not helpful.

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**